libraries/chain/market_evaluator.cpp

```diff
@@ -237,18 +238,20 @@ void_result call_order_update_evaluator::do_apply(const call_order_update_operat
                                              _bitasset_data->current_feed.maintenance_collateral_ratio);
          call.target_collateral_ratio = o.extensions.value.target_collateral_ratio;
       });
+      call_order_id = call_obj->id;
    }
    else // updating existing debt position
    {
       call_obj = &*itr;
       auto new_collateral = call_obj->collateral + o.delta_collateral.amount;
       auto new_debt = call_obj->debt + o.delta_debt.amount;
+      call_order_id = call_obj->id;
 
       if( new_debt == 0 )
       {
          FC_ASSERT( new_collateral == 0, "Should claim all collateral when closing debt position" );
          d.remove( *call_obj );
-         return void_result();
+         return call_order_id;
```

Comment:
Should we return the id of a deleted object?

Reply:
Agree, makes sense to return it.